hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/contracts/operations/HederaBalanceOperation.java

```diff
@@ -55,20 +36,32 @@
 public class HederaBalanceOperation extends BalanceOperation {
 
     private BiPredicate<Address, MessageFrame> addressValidator;
+    private final Predicate<Address> systemAccountDetector;
 
-    public HederaBalanceOperation(GasCalculator gasCalculator, BiPredicate<Address, MessageFrame> addressValidator) {
+    public HederaBalanceOperation(
+            GasCalculator gasCalculator,
+            BiPredicate<Address, MessageFrame> addressValidator,
+            Predicate<Address> systemAccountDetector) {
         super(gasCalculator);
         this.addressValidator = addressValidator;
+        this.systemAccountDetector = systemAccountDetector;
     }
 
     @Override
     public OperationResult execute(MessageFrame frame, EVM evm) {
+        final Supplier<OperationResult> systemAccountExecutionSupplier = () -> {
+            frame.popStackItems(1); // clear the address from the stack
```

Comment:
No, this is the normal flow of the operation, except we are substituting execution in the address checker if we have a system account. The top item contains the address we are querying, which would have been popped in normal execution, so we need to pop it as part of the operation since we are not doing normal exeution,